source/common/redis/command_splitter_impl.cc

```diff
@@ -52,10 +53,17 @@ void SimpleRequest::cancel() {
   handle_ = nullptr;
 }
 
+MGETRequest::~MGETRequest() {
+#ifndef NDEBUG
+  for (const PendingRequest& request : pending_requests_) {
```

Comment:
I'd be curious if the compiler optimizer is smart enough to remove the entire loop when ASSERT is compiled out rendering this #ifdef unnecessary.